### PR TITLE
[MO][OPS-12015] Redirect to request received page after failing HMRC …

### PR DESCRIPTION
**Pull request by Lyrain**
…checks

### CODE REVIEW

File: app/config/AppConfig.scala [hunk 1]
```diff
@@ -36,7 +36,9 @@ class AppConfig @Inject() (servicesConfig: ServicesConfig, configuration: Config
     val failedAttemptRepoMaxAttempts: Int = configuration.get[Int]("mongodb.failed-attempts.failed-attempt-repo-max-attempts")
   }
 
-  val platformFrontendHost: String = readConfigAsValidUrlString("platform.frontend.host")
+  val platformHost: Option[String] = configuration.getOptional[String]("platform.frontend.host")
+  val platformFrontendHost: String = readStringAsValidUrlString(platformHost.getOrElse(configuration.get[String]("urls.localhost")))
+  val feedbackFrontendUrl: String = readStringAsValidUrlString(platformHost.getOrElse(configuration.get[String]("urls.feedback-frontend")))
```
Comment (Lyrain) [inline app/config/AppConfig.scala:41]: Improvement for changes made in OPS-11532. Makes it so we don't need to point to production in all environments.

File: app/controllers/WeAreVerifyingYourBankAccountController.scala [hunk 2]
```diff
@@ -68,27 +71,43 @@ class WeAreVerifyingYourBankAccountController @Inject() (
       isValidEventValue:             EventValue,
       bankDetailsRiskResultResponse: GetBankDetailsRiskResultResponse,
       bankAccountSummary:            BankAccountSummary,
-      consentStatus:                 Option[ConsentStatus]
-  )(implicit request: RequestHeader): Future[(Result, Journey)] = isValidEventValue match {
-    case EventValue.NotReceived => Future.successful(
+      consentStatus:                 ConsentStatus
+  )(implicit request: RequestHeader): Future[(Result, Journey)] = (isValidEventValue, consentStatus) match {
+    case (_, ConsentStatus.Failed) => Future.successful {
```
Comment (Lyrain) [inline app/controllers/WeAreVerifyingYourBankAccountController.scala:76]: This is the nicest way I could think to do this for AC2

File: app/models/journeymodels/HasFinished.scala [hunk 3]
```diff
@@ -41,6 +42,11 @@ object HasFinished {
    */
   case object YesSucceeded extends HasFinished
 
+  /**
+   * Journey has finished in successful way. The refund has been sent to case management for manual review.
+   */
+  case object YesSentToCaseManagement extends HasFinished
```
Comment (Lyrain) [inline app/models/journeymodels/HasFinished.scala:48]: Added separate state for a Ecospend success + EDH risk fail.

File: app/config/AppConfig.scala [hunk 4]
```diff
@@ -93,6 +93,13 @@ class AppConfig @Inject() (servicesConfig: ServicesConfig, configuration: Config
     )
   }
 
+  private def readStringAsValidUrlString(url: String): String = {
```
Comment (pawelpanasewicz) [inline app/config/AppConfig.scala:96]: I'd also add a key here and try to read the config here. So, when throwing an exception, we would know the config key that was misconfigured

File: test/pagespecs/VerifyBankAccountPageSpec.scala [hunk 5]
```diff
@@ -206,25 +206,26 @@ class VerifyBankAccountPageSpec extends ItSpec {
     NpsClaimOverpaymentStub.verifyNoneClaimOverpayment(tdAll.nino, tdAll.p800ReferenceSanitised)
   }
 
-  "redirect to 'Request not submitted' page when EDH call results in nextAction=DoNotPay" in {
+  "redirect to 'Request received' page when EDH call results in nextAction=DoNotPay" in {
+    // NOTE: When the EDH risk call fails, the application should direct the user to the 'Request received' page.
```
Comment (barnesjake) [inline test/pagespecs/VerifyBankAccountPageSpec.scala:210]: Maybe just add to the comment to say what the ticket mentions: 
`were the customer passes the EcoSpend fraud algorithm but fails the RIS Risking/EDH, we need to mirror the PTA and advise the customer the refund is being processed`

File: conf/application.conf [hunk 6]
```diff
@@ -89,6 +87,8 @@ urls {
     lost-national-insurance-number = "https://www.gov.uk/lost-national-insurance-number"
   }
 
+  localhost = "http://localhost:10150"
+
   feedback-frontend = "http://localhost:9514"
```
[the remark this answers is not in the document]
Comment (Lyrain) [reply conf/application.conf:92]: Oh, I didn't know about HOCON defaults. That makes this much simpler.

File: app/config/AppConfig.scala [hunk 7]
```diff
@@ -87,7 +86,7 @@ class AppConfig @Inject() (servicesConfig: ServicesConfig, configuration: Config
    */
   private def readConfigAsValidUrlString(configPath: String): String = {
     val url: String = configuration.get[String](configPath)
-    Try(new java.net.URI(url)).fold[String](
+    Try(new java.net.URI(url).toURL).fold[String](
```
Comment (Lyrain) [inline app/config/AppConfig.scala:89]: Discovered on accident, java.net.URI doesn't do the same validation as java.net.URL but can't use the former constructor as it's deprecated.

**Review by pawelpanasewicz — APPROVED** [review]

**Pull request merged**